src/render/software/SDL_triangle.c

```diff
@@ -564,17 +576,6 @@ int SDL_SW_BlitTriangle(
         has_modulation = SDL_TRUE;
     }
 
-    {
-        /* Clip triangle rect with surface rect */
-        SDL_Rect rect;
-        rect.x = 0;
-        rect.y = 0;
-        rect.w = dst->w;
-        rect.h = dst->h;
-
-        SDL_GetRectIntersection(&dstrect, &rect, &dstrect);
-    }
-
```

Comment:
this is about the dst rect, not sure we want to clamp it?

Reply:
The clipping rect defaults to the full surface, so the second intersection covers everything.